src/main/java/com/shinyhut/vernacular/client/VernacularClient.java

```diff
@@ -47,16 +47,26 @@ public VernacularClient(VernacularConfig config) {
      * @param port Remote port to connect to
      * @throws IllegalStateException if the client is already running
      */
-    public void start(String host, int port) {
+    public void start(String host, int port) throws IOException {
+        startSocket(new Socket(host, port));
+    }
+
 
+    /**
+     * Starts the VNC client by connecting to the specified socket
+     *
+     * @param socket Socket to connect to
+     * @throws IllegalStateException if the client is already running
+     */
+    public void startSocket(Socket socket) {
```

Comment:
Let's call this method `public void start(Socket socket)` - the parameter type makes it clear what it's connecting to,

Reply:
Got that